### PR TITLE
Adds extra tests for CaseInsensitiveDict

### PR DESCRIPTION
**Pull request by gvangool**
This adds extra tstes for `requests.structures.CaseInsensitiveDict`

- Test for NotImplemented in `__eq__`
- Adds test for `copy()`
- Adds test for `__repr__()`

### CODE REVIEW

File: test_requests.py [hunk 1]
```diff
@@ -1224,6 +1224,7 @@ def test_equality(self):
         del othercid['spam']
         assert cid != othercid
         assert cid == {'spam': 'blueval', 'eggs': 'redval'}
+        assert cid.__eq__(object()) == NotImplemented
```
Comment (sigmavirus24) [inline test_requests.py]: Shouldn't this be

``` py
with pytest.raises(NotImplementedError):
    cid == object()
```

We shouldn't be directly invoking dunder methods anywhere in the code base, not even tests.

File: test_requests.py [hunk 2]
```diff
@@ -1261,6 +1262,20 @@ def test_preserve_last_key_case(self):
...
         assert frozenset(cid) == keyset
 
+    def test_copy(self):
+        cid = CaseInsensitiveDict({
+            'Accept': 'application/json',
+            'user-Agent': 'requests',
+        })
+        assert cid == cid.copy()
+
+    def test_repr(self):
```
Comment (sigmavirus24) [inline test_requests.py:1275]: I, for one, do not agree with testing reprs. @Lukasa thoughts?

File: test_requests.py (hunk 1 above)
Comment (gvangool) [inline test_requests.py]: Except that it's not raised, but returned. So not sure how else to test this.

Comment (sigmavirus24) [inline test_requests.py]: If you have to access private methods to test something, you shouldn't be testing it.

I thought that returning `NotImplemented` triggered a `NotImplementedError` but that doesn't happen in all cases it seems.

``` py
cid == object()
# => False
```

But I don't think we strictly need this assertion anyway. You're testing an implementation detail, not a public part of the API (which is all we should ever test).

File: test_requests.py (hunk 2 above)
Comment (Lukasa) [inline test_requests.py:1275]: I only ever test reprs when I think they're an important part of the API. In this case I don't really mind.

File: test_requests.py (hunk 1 above)
Comment (gvangool) [inline test_requests.py]: @sigmavirus24 I've changed it in 47e71d1774745543b743bdf570d9d8e7151355e6

**Comment by sigmavirus24** [comment]
@Lukasa how does this look to you? Looks fine to me. I don't mind a couple extra tests personally.

File: test_requests.py [hunk 3]
```diff
@@ -1261,6 +1262,20 @@ def test_preserve_last_key_case(self):
         assert frozenset(cid.keys()) == keyset
         assert frozenset(cid) == keyset
 
+    def test_copy(self):
+        cid = CaseInsensitiveDict({
+            'Accept': 'application/json',
+            'user-Agent': 'requests',
+        })
+        assert cid == cid.copy()
```
Comment (Lukasa) [inline test_requests.py]: Should this test also assert that the copy does not lead to two objects that are literally the same? e.g., should we also add `assert cid is not cid.copy()`?

**Comment by Lukasa** [comment]
I have one note, but I'm happy to add in extra tests.

File: test_requests.py (hunk 3 above)
Comment (gvangool) [inline test_requests.py]: I believe I fixed this in 3b325f1bc5df476c4c4e93c5a742d81e2c5bd950

**Comment by gvangool** [comment]
@Lukasa I believe I've addressed your concern for that :)

**Comment by Lukasa** [comment]
Yeah, that looks good to me. \o/

**Pull request merged**